apps/builder/app/builder/features/sidebar-left/sidebar-left.tsx

```diff
@@ -32,6 +32,20 @@ const useActiveTab = () => {
   return [nextTab, $activeSidebarPanel.set] as const;
 };
 
+const useHideActiveTabOnPreview = () => {
+  const previousRef = useRef($isPreviewMode.get());
+
+  useEffect(() => {
+    return $isPreviewMode.subscribe((isPreviewMode) => {
+      // When user switches to preview mode we want to hide any active sidebar panel.
+      if (isPreviewMode && previousRef.current === false) {
```

Comment:
why check previous state and not just always set none?

Reply:
because I need to know if the change was  from preview mode to builder or from builder to preview ... otherwise I would be hiding everything also when switch happens to the builder .... 

actually as I wrote this I realized I can just check if it was already switched lol